sensor/kubernetes/crs/crs.go

```diff
@@ -0,0 +1,299 @@
+package crs
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	metautils "github.com/grpc-ecosystem/go-grpc-middleware/v2/metadata"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/booleanpolicy/policyversion"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/crs"
+	"github.com/stackrox/rox/pkg/env"
+	grpcUtil "github.com/stackrox/rox/pkg/grpc/util"
+	"github.com/stackrox/rox/pkg/k8sutil"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/pkg/pods"
+	protoconv "github.com/stackrox/rox/pkg/protoconv/certs"
+	"github.com/stackrox/rox/pkg/protoutils"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/version"
+	"github.com/stackrox/rox/sensor/common/centralclient"
+	sensorCommon "github.com/stackrox/rox/sensor/common/sensor"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh/securedcluster"
+	"github.com/stackrox/rox/sensor/kubernetes/helm"
+	"github.com/stackrox/rox/sensor/kubernetes/sensor"
+	"google.golang.org/grpc/metadata"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	log                                  = logging.LoggerForModule()
+	fetchSensorDeploymentOwnerRefBackoff = wait.Backoff{
+		Duration: 10 * time.Millisecond,
+		Factor:   3,
+		Jitter:   0.1,
+		Steps:    10,
+		Cap:      6 * time.Minute,
+	}
+)
+
+// EnsureServiceCertificatesPresent initiates the CRS-based cluster registration flow to retrieve the
+// service certificates in case no services certificates are found and instead a CRS is present.
+func EnsureServiceCertificatesPresent() error {
+	log.Infof("Ensuring certificates for Secured Cluster services are present.")
+	clientconn.SetUserAgent(fmt.Sprintf("%s Cluster Registration Helper", clientconn.Sensor))
+
+	// Check if legacy sensor service certificate, e.g. created via init-bundle, exists.
+	if legacySensorServiceCert := os.Getenv(crs.LegacySensorServiceCertEnvName); legacySensorServiceCert != "" {
+		log.Infof("Legacy sensor service certificate available, skipping cluster registration using Cluster Registration Secret.")
+		return nil
+	}
+
+	// Check if modern sensor service certificate, e.g. created by CRS-based cluster registration or
+	// refreshed by periodic cert rotation, exists.
+	if sensorServiceCert := os.Getenv(crs.SensorServiceCertEnvName); sensorServiceCert != "" {
```

Comment:
So the plan is:
1. this code to run as the first init container, and directly mount the secrets,
2. then the next init container is `init-tls-certs` from @vladbologa's PR,
3. And then the regular sensor runs, mounting the emptydir with just one set of certs

right?

Reply:
yes.